docs2/site/docs/getting-started/custom-scalars.md

```diff
@@ -183,105 +213,303 @@ public class ExampleSchema : Schema
+    public override object ParseLiteral(IValue value) => value switch
+    {
+        StringValue s => int.TryParse(s.Value, out int i) && i > 0 ? i : throw new FormatException($"'{s.Value}' is not a valid identifier."),
+        null => 0,
+        _ => ThrowLiteralConversionError(value)
+    };
 
-- Since GraphQL specifies no response format, `Serialize` is not responsible for preparing the scalar for transport to the client. It is only responsible for generating an object which can eventually be serialized by `IDocumentWriter` or other transport-focused API.
+    public override object ParseValue(object value) => value switch
+    {
+        string s => int.TryParse(s, out int i) && i > 0 ? i : throw new FormatException($"'{s}' is not a valid identifier."),
+        null => 0,
+        _ => ThrowValueConversionError(value)
+    };
 
-In this example, you created a custom scalar. In summary:
+    public override object Serialize(object value) => value switch
+    {
+        int i => i > 0 ? i.ToString() : i == 0 ? null : ThrowSerializationError(value),
+        _ => ThrowSerializationError(value)
+    };
+}
+```
 
-- Create a class for the server-side representation of the scalar
-- Implement a `ScalarGraphType` which handles parsing, literal parsing, and serialization
-- Register the `ScalarGraphType with the DI container
-- Define how to parse the raw representation of the scalar to its server-side representation using `ValueConverter.Register`
-- Implement a `ValueNode<T>` class for the server-side representation
-- Implement an `IAstFromValueConverter` for wrapping the server-side representation in its `ValueNode<T>` implementation
-- Register the `IAstFromValueConverter` with the schema using `Schema.RegisterValueConverter`
+## Replacing built-in scalar types
+
+In some cases you may want or need to replace the functionality of the built-in graph types. This can
+be accomplished by running a schema node visitor on the schema after it has been initialized. Keep in
+mind that the node visitor will walk through the introspection portion of the schema, so replacing
+a built-in type may affect the operation of introspection queries.
+
+Below is a sample of how to replace the built-in `IdGraphType` with the `DbIdGraphType` shown above.
+
+1. Name the `DbIdGraphType` to be the same as the built-in type
+
+```csharp
+public class DbIdGraphType : ScalarGraphType
+{
+    public DbIdGraphType()
+    {
+        Name = "ID";
+    }
+
+    ...
+}
+```
+
+2. Create a node walker
+
+```csharp
+// sungam3r to provide sample
+// perhaps we should provide a node walker for this purpose, or provide a more convenient method to replace types
+// note: don't we need to fix up schematypes also so that if a type name is provided to a query variable argument, it gets a proper reference?
```

Comment:
This is an interesting question. Perhaps not because _replacing_  involves replacement of behavior but not name. Name should stay the same.

Reply:
We are testing for this scenario with the solution in #2388 as it tests variables, literals, and serialization, for both schema-first and code-first.

Reply:
So the names match. Right?

Reply:
Right.

Reply:
Sample replaced, so resolving this not-applicable comment.